src/lib/net.py

```diff
@@ -23,22 +22,35 @@
 from .utils import ROOT
 
 
-def download_snapshot(url: Union[Path, str], ext: str = None, offline: bool = False) -> str:
+def download_snapshot(url: str, output_folder: Path, ext: str = None, offline: bool = False) -> str:
     """
     This function downloads a file into the snapshots folder and outputs the
     hashed file name based on the input URL. This is used to ensure
     reproducibility in downstream processing, which will not require to network
     access.
+
+    Args:
+        url: URL to download a resource from
+        output_folder: Root folder where snapshot, intermediate and tables will be placed.
+        ext: Force extension when creating output file, handy when it cannot be guessed from URL.
+        offline: If true, skip download and simply return the deterministic path where this file
+            would have been downloaded.
+
+    Returns:
+        str: Absolute path where this file was downloaded. This is a deterministic output; the same
+            URL will always produce the same output path.
     """
-    url = str(url)
+
+    # Create the snapshots folder if it does not exist
+    (output_folder / "snapshot").mkdir(parents=True, exist_ok=True)
+
+    # Create a deterministic file name
     if ext is None:
         ext = url.split(".")[-1]
-    file_path = (
-        ROOT / "output" / "snapshot" / ("%s.%s" % (uuid.uuid5(uuid.NAMESPACE_DNS, url), ext))
-    )
+    file_path = output_folder / "snapshot" / ("%s.%s" % (uuid.uuid5(uuid.NAMESPACE_DNS, url), ext))
 
-    # Download the file if online
-    if not offline:
+    # Skip download of the file if offline flag is present or if the file does not exist
+    if not offline or not file_path.exists():
```

Comment:
I didn't word that correctly, PTAL. Done.

Reply:
When I initially saw the term 'offline' I thought that it was for use when offline, and would not require network access.  Looking at this closer I realize that it is ignored if the file does not already exist, so in the event that the file does not already exist it would still fail when offline even with offline set to true; this really just controls whether the file is to be replaced if it already exists.

So consider using an argument name like "skip_overwrite" or "skip_if_exists", "skip_existing", or else reversing the value and calling it something like "replace" or "force_replace".